rest/data/host_create.go

```diff
@@ -151,6 +151,64 @@ func createHostFromCommand(cmd model.PluginCommandConf) (*apimodels.CreateHost,
 }
 
 func (dc *DBCreateHostConnector) MakeIntentHost(taskID, userID, publicKey string, createHost apimodels.CreateHost) (*host.Host, error) {
+	if createHost.CloudProvider == evergreen.ProviderNameDocker {
+		return dc.makeDockerIntentHost(userID, createHost)
+	}
+	return dc.makeEC2IntentHost(taskID, userID, publicKey, createHost)
+}
+
+func (dc *DBCreateHostConnector) makeDockerIntentHost(userID string, createHost apimodels.CreateHost) (*host.Host, error) {
+	d := distro.Distro{}
+	var err error
+
+	distroID := createHost.Distro
+	if distroID != "" {
+		d, err = distro.FindOne(distro.ById(distroID))
+		if err != nil {
+			return nil, errors.Wrapf(err, "error finding distro %s", distroID)
+		}
+	} else {
+		return nil, errors.New("distro id cannot be empty")
+	}
+
+	options := cloud.HostOptions{}
+	if userID != "" {
+		options.UserName = userID
+		options.UserHost = true
+	}
+	options.DockerOptions = host.DockerOptions{
+		Image:            createHost.Image,
+		Command:          createHost.Command,
+		RegistryName:     createHost.Registry.Name,
+		RegistryUsername: createHost.Registry.Username,
+		RegistryPassword: createHost.Registry.Password,
+	}
+
+	if d.ContainerPool != "" {
```

Comment:
No need to do this block - just insert it and let a job that you will implement later handle finding a parent